sentry-android-core/src/main/java/io/sentry/android/core/AndroidContinuousProfiler.java

```diff
@@ -103,7 +104,10 @@ private void init() {
             logger);
   }
 
-  public synchronized void start(final @NotNull TracesSampler tracesSampler) {
+  @Override
+  public synchronized void startProfileSession(
```

Comment:
AFAIK we should avoid using synchronized, and use lock objects in favor instead (e.g. see https://github.com/getsentry/sentry-java/blob/f1b71160573bb0a81119da99a4b995b5a03d47b4/sentry-android-core/src/main/java/io/sentry/android/core/ActivityFramesTracker.java#L141)